templates/templates/BenchmarkDotNet.BenchmarkProjectTemplate.CSharp/_BenchmarkProjectName_.csproj

```
@@ -1,7 +1,11 @@
 <Project Sdk="Microsoft.NET.Sdk">
-  <PropertyGroup>
-    <TargetFrameworks>$(Frameworks)</TargetFrameworks>
+  <PropertyGroup Condition="'$(consoleApp)' == 'true'">
+    <TargetFramework>$(Frameworks)</TargetFramework>
+    <OutputType>Exe</OutputType>
   </PropertyGroup>
+  <PropertyGroup Condition="'$(consoleApp)' == 'false'">
+    <TargetFrameworks>$(Frameworks)</TargetFrameworks>
+ </PropertyGroup>
   <PropertyGroup>
     <PlatformTarget>AnyCPU</PlatformTarget>
```

Should we always have a single target framework?

```suggestion
  <PropertyGroup Condition="'$(consoleApp)' == 'true'">
    <OutputType>Exe</OutputType>
  </PropertyGroup>
  <PropertyGroup>
    <TargetFramework>$(Frameworks)</TargetFramework>
    <PlatformTarget>AnyCPU</PlatformTarget>
```

---

hi @adamsitnik , it could be multiple target frameworks.
However, `dotnet run` will throw error without the framework option if there were multiple target frameworks, which may not be ideal for user experience. 
Plus, I guess it will be rare that developers want to have multiple target frameworks when building a console app to run benchmark. unless I was wrong?

---

Can user specify multiple target frameworks when creating a new project from template? If not, we should simplify it and users can always modify it later.

---

I agree with @adamsitnik. Templates like `dotnet new console` or `dotnet new classlib` (and the projects created by VS templates) just offer `<TargetFramework>` (singular). Users can choose to modify project files based on requirements.

ps: on the topic of keeping things simple, it would be very intuitive for `dotnet` cli users  if `dotnet new benchmark.console` and `dotnet new benchmark.classlib` are separated, rather than expecting `--console-app true` with classlib variant as default. At the very least, we should prefer `--console` flag (presence means yes, absence means no) instead of two-worded option with mandatory boolean value. (we are still in version 0.x, so breaking changes will not be frowned upon by end users)

---

Hi @adamsitnik, I think for the classlib, it supports multiple templates, and some user would choose to do so. 
but as @am11 mentioned, the `console` and `classlib` ones from Microsoft generates `TargetFramework`. 
I will made the change, all with single TargetFramework only. 
And I think it would be better to address the address the suggestion from @am11 on separate templates in different pull request.

---

hi @adamsitnik  and @am11 , template are updated with single `TargetFramework` for both console and classlib per discussion.